### ISSUE TITLE
License?

### ISSUE THREAD

**Issue by feklee** [trinitronx/vncpasswd.py#2]
I would like to try porting `d3des.py` to Python 3 and include it with [vncdesk][1].

The disclaimer in `README.md` quotes the MIT license, but it is augmented by a phrase, a restriction which would make *vncpassword.py* incompatible with most open source licenses including the GPL: "Please only use this software for Good, and don't harm directly or indirectly living beings"

*Is that phrase part of the license under which `d3des.py` is released?*

Sorry for license issues. I just want to avoid limiting distribution of vncdesk (which is released under the WTFPL, but I could change to MIT - that would be OK).

[1]: https://github.com/feklee/vncdesk

**Comment by trinitronx** [comment]
There was some confusion about the licensing of `d3des.py` as written by Yusuke Shinyama [as discussed in this RedHat bug](https://bugzilla.redhat.com/show_bug.cgi?id=567877).

Basically it all boiled down to this:

> Since the original d3des code is Public Domain, and this Python code is a derived work of that, as opposed to being a derived work of the VNC viewer code, the license is fine.

This file was lifted directly from the [Python `vnc2flv`](http://pydoc.net/Python/vnc2flv/20100207/vnc2flv.d3des/) source code. As such, it's already in public domain and is simply an open source dependency of my code.  Therefore, you can lift it direct from the source without worries of GPL incompatibility ;-)

Basically since my tool is for decrypting VNC passwords, I put that extra clause in there to guard against misuse.

**Comment by feklee** [comment]
Thanks for pointing me to [Redhat Bug 567877][1]! Not sure about public domain, though. Yusuke Shinyama put the vnc2flv package under an MIT license. According to the package [meta data][2]:

> License: MIT/X

Anyhow, MIT is fine for me too, and anyhow I also contacted Yusuke Shinyama.

[1]: https://bugzilla.redhat.com/show_bug.cgi?id=567877
[2]: http://pydoc.net/Python/vnc2flv/20100207/
